code/ai/aicode.cpp

```diff
@@ -2495,33 +2502,38 @@ int get_enemy_timestamp()
 /**
  * Return objnum if enemy found, else return -1;
  *
- * @param objnum		Object number
- * @param range			Range within which to look
- * @param max_attackers Don't attack a ship that already has at least max_attackers attacking it.
+ * @param objnum         Object number
+ * @param range          Range within which to look
+ * @param max_attackers  Don't attack a ship that already has at least max_attackers attacking it.
+ * @param ship_info_index  If specified, restrict the search to enemies with this ship class
+ * @param class_type     If specified, restrict the search to enemies with this ship type
  */
-int find_enemy(int objnum, float range, int max_attackers, int ship_info_index)
+int find_enemy(int objnum, float range, int max_attackers, int ship_info_index, int class_type)
 {
-	int	enemy_team_mask;
+	int enemy_team_mask;
 
 	if (objnum < 0)
 		return -1;
 
 	enemy_team_mask = iff_get_attackee_mask(obj_team(&Objects[objnum]));
 
 	//	if target_objnum != -1, use that as goal.
-	ai_info	*aip = &Ai_info[Ships[Objects[objnum].instance].ai_index];
+	ai_info* aip = &Ai_info[Ships[Objects[objnum].instance].ai_index];
 	if (timestamp_elapsed(aip->choose_enemy_timestamp)) {
 		aip->choose_enemy_timestamp = timestamp(get_enemy_timestamp());
 		if (aip->target_objnum != -1) {
-			int	target_objnum = aip->target_objnum;
+			int target_objnum = aip->target_objnum;
+			ship* target_shipp = &Ships[Objects[target_objnum].instance];
```

Comment:
Change this to:
```C++
ship* target_shipp = (Objects[target_objnum].type == OBJ_SHIP) ? &Ships[Objects[target_objnum].instance] : nullptr;
```
Because not every target is necessarily a ship.  This was a bug in Volition's code too :-/